Sources/Noora/Noora.swift

```diff
@@ -480,10 +480,10 @@ extension Noorable {
         )
     }
 
-    public func progressStep(
+    public func progressStep<V>(
         message: String,
-        task: @escaping ((String) -> Void) async throws -> Void
-    ) async throws {
+        task: @escaping ((String) -> Void) async throws -> V
+    ) async throws -> V {
```

Comment:
What would be the experience if you are not interested in returning any value? I'd not expect any warning nor having to return something to match the API.